### ISSUE TITLE
Error and not loading since upgrade to 0.2

### ISSUE THREAD

**Issue by Somapolice** [uvjustin/alarmdotcom#70]
Logger: homeassistant.config_entries
Source: config_entries.py:344
First occurred: 11:24:28 AM (2 occurrences)
Last logged: 11:25:10 AM

Config entry 'None' for alarmdotcom integration not ready yet; Retrying in background

I use Brinks via Alarm.com. Was working fine with old Config.

**Comment by RB14060** [comment]
Having the same issue on my setup. Component is just stuck in "Retrying setup" status.

**Comment by elahd** [comment]
Thanks for reporting. Try this:

1. Update to the latest beta (v0.2.0-beta.2). (The newest beta spits out more verbose logs.)
2. Add the below to configuration.yaml:
```yaml
logger:
  default: warn
  logs:
    custom_components.alarmdotcom: debug
    pyalarmdotcomajax: debug
```
3. Restart Home Assistant.

If you run into an problem, check the Home Assistant logs for anything relayed to `alarmdotcom` or `pyalarmdotcomajax`.

**Comment by RB14060** [comment]
Just upgraded and this is what I'm getting now. Not sure if the VIEWSTATE strings have any sensitive info in them or not so I masked them just to be safe, I can provide the full output if necessary. It did properly recognize my provider and pull the User ID as well.

> 2022-01-24 11:36:10 DEBUG (MainThread) [custom_components.alarmdotcom] custom_components.alarmdotcom: Initializing Alarmdotcom from config entry.
> 2022-01-24 11:36:10 DEBUG (MainThread) [custom_components.alarmdotcom.controller] custom_components.alarmdotcom.controller: Setting up controller with config_entry: {'username': 'XXXXX', 'password': 'XXXXX', '2fa_cookie': None}
> 2022-01-24 11:36:10 DEBUG (MainThread) [pyalarmdotcomajax] Attempting to log in to Alarm.com
> 2022-01-24 11:36:15 DEBUG (MainThread) [pyalarmdotcomajax] Response status from Alarm.com: 200
> 2022-01-24 11:36:15 DEBUG (MainThread) [pyalarmdotcomajax] {'__VIEWSTATE': 'XXXXX', '__VIEWSTATEGENERATOR': 'C2EE9ABB', '__EVENTVALIDATION': 'XXXXX', '__PREVIOUSPAGE': 'XXXXX'}
> 2022-01-24 11:36:15 INFO (MainThread) [pyalarmdotcomajax] Attempting login to Alarm.com
> 2022-01-24 11:36:15 DEBUG (MainThread) [pyalarmdotcomajax] Got Provider: XXXXX, User ID: XXXXX
> 2022-01-24 11:36:15 DEBUG (MainThread) [pyalarmdotcomajax] Calling update on Alarm.com
> 2022-01-24 11:36:15 DEBUG (MainThread) [pyalarmdotcomajax.entities] Initialized ADCRelationshipType.SYSTEM (systems/system) <MY ADDRESS>
> 2022-01-24 11:36:15 DEBUG (MainThread) [pyalarmdotcomajax.entities] Initialized ADCRelationshipType.PARTITION (devices/partition) Panel
> 2022-01-24 11:36:15 DEBUG (MainThread) [pyalarmdotcomajax.entities] Initialized ADCRelationshipType.SENSOR (devices/sensor) Back Door
> 2022-01-24 11:36:15 DEBUG (MainThread) [pyalarmdotcomajax.entities] Initialized ADCRelationshipType.SENSOR (devices/sensor) Downstairs Smoke
> 2022-01-24 11:36:15 DEBUG (MainThread) [pyalarmdotcomajax.entities] Initialized ADCRelationshipType.SENSOR (devices/sensor) Front Door
> 2022-01-24 11:36:15 DEBUG (MainThread) [pyalarmdotcomajax.entities] Initialized ADCRelationshipType.SENSOR (devices/sensor) Garage Door
> 2022-01-24 11:36:15 DEBUG (MainThread) [pyalarmdotcomajax.entities] Initialized ADCRelationshipType.SENSOR (devices/sensor) Garage Window
> 2022-01-24 11:36:15 DEBUG (MainThread) [pyalarmdotcomajax.entities] Initialized ADCRelationshipType.SENSOR (devices/sensor) Kitchen Window
> 2022-01-24 11:36:15 DEBUG (MainThread) [pyalarmdotcomajax.entities] Initialized ADCRelationshipType.SENSOR (devices/sensor) Library Window
> 2022-01-24 11:36:15 DEBUG (MainThread) [pyalarmdotcomajax.entities] Initialized ADCRelationshipType.SENSOR (devices/sensor) Living Room Motion
> 2022-01-24 11:36:15 DEBUG (MainThread) [pyalarmdotcomajax.entities] Initialized ADCRelationshipType.SENSOR (devices/sensor) Living Room Window 1
> 2022-01-24 11:36:15 DEBUG (MainThread) [pyalarmdotcomajax.entities] Initialized ADCRelationshipType.SENSOR (devices/sensor) Living Room Window 2
> 2022-01-24 11:36:15 DEBUG (MainThread) [pyalarmdotcomajax.entities] Initialized ADCRelationshipType.SENSOR (devices/sensor) Panel Camera
> 2022-01-24 11:36:15 DEBUG (MainThread) [pyalarmdotcomajax.entities] Initialized ADCRelationshipType.SENSOR (devices/sensor) Stair Window
> 2022-01-24 11:36:15 DEBUG (MainThread) [pyalarmdotcomajax.entities] Initialized ADCRelationshipType.SENSOR (devices/sensor) Upstairs Smoke Detector
> 2022-01-24 11:36:15 DEBUG (MainThread) [pyalarmdotcomajax] Failed to get data for device type ADCDeviceType.GARAGE_DOOR. Response: [{'status': '423', 'detail': 'You do not have permissions for this device type', 'code': 423}]
> 2022-01-24 11:36:15 DEBUG (MainThread) [pyalarmdotcomajax] pyalarmdotcomajax: Showing first error only. Status: 423. Response: {'included': [], 'errors': [{'status': '423', 'detail': 'You do not have permissions for this device type', 'code': 423}], 'meta': {'transformer_version': '1.1'}}
> 2022-01-24 11:36:15 ERROR (MainThread) [custom_components.alarmdotcom.controller] custom_components.alarmdotcom.controller: get_controller failed with CannotConnect exception:
> 2022-01-24 11:36:15 ERROR (MainThread) [custom_components.alarmdotcom.controller] custom_components.alarmdotcom.controller: get_controller failed with CannotConnect exception:
> 2022-01-24 11:36:15 WARNING (MainThread) [homeassistant.config_entries] Config entry 'None' for alarmdotcom integration not ready yet; Retrying in background

**Comment by elahd** [comment]
Thanks, @RB14060. Do you have a garage door connected to Alarm.com?

**Comment by RB14060** [comment]
> Thanks, @RB14060. Do you have a garage door connected to Alarm.com?

No, I don't.

**Comment by elahd** [comment]
Thanks for that. Try this:

1. Delete the integration from your integrations page.
2. Update the integration to v0.2.0-beta.3
3. Restart Home Assistant
4. Try setting up the integration again.

**Comment by RB14060** [comment]
Working now, configured properly and I got the area assignment screen with all my sensors and such. Thanks!

**Comment by Somapolice** [comment]
Fixed with beta 3

**Comment by Somapolice** [comment]
Fixed with beta 3

**Comment by AdventureAhead** [comment]
Thank you, the integration now configures successfully with beta.3.
